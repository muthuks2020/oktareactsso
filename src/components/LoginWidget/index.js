import React, { useState } from "react";
import PropTypes from 'prop-types';
import OktaAuth from "@okta/okta-auth-js";
import { useOktaAuth } from "@okta/okta-react";
import {
  Button,
  Grid,
  TextField,
  Container,
  CircularProgress,
  Typography,
  Fade,
} from "@material-ui/core";
import CssBaseline from "@material-ui/core/CssBaseline";
import { Alert } from "components";
import { useWidgetStyles } from "./styles";

const LoginWidget = ({ issuer }) => {
  const classes = useWidgetStyles();
  const { authService } = useOktaAuth();
  const [error, setError] = useState();
  const [loader, setLoader] = useState(false);
  const [username, setUsername] = useState();
  const [password, setPassword] = useState();

  const handleSubmit = async(e) => {
    e.preventDefault();
    setLoader(true);
    const oktaAuth = new OktaAuth({ issuer });

    try {
      const { sessionToken } = await oktaAuth.signIn({ username, password });
      await authService.redirect({ sessionToken });
      return null
    } catch (err) {
      handleError(err);
    }
  };

  const handleError = (err) => {
    setLoader(false);
    setError(err);
  };

  const handleUsernameChange = (e) => {
    setUsername(e.target.value);
  };

  const handlePasswordChange = (e) => {
    setPassword(e.target.value);
  };

  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
        <Typography component="h1" variant="h5">
          {loader ? <CircularProgress /> : "Sign in"}
        </Typography>
        <form className={classes.form} onSubmit={handleSubmit} noValidate>
          <TextField
            variant="outlined"
            margin="normal"
            required
            fullWidth
            id="username"
            label="Username"
            name="username"
            autoComplete="username"
            onChange={handleUsernameChange}
            autoFocus
          />
          <TextField
            variant="outlined"
            margin="normal"
            required
            fullWidth
            name="password"
            label="Password"
            type="password"
            id="password"
            onChange={handlePasswordChange}
            autoComplete="current-password"
          />
          <Button
            id="submit"
            type="submit"
            value="Submit"
            fullWidth
            variant="contained"
            className={classes.submit}
          >
            Log in
          </Button>
          {error && (
            <Grid item>
              <Fade in>
                <Alert severity="error">{error.message}</Alert>
              </Fade>
            </Grid>
          )}
        </form>
      </div>
    </Container>
  );
};

LoginWidget.propTypes = {
  issuer: PropTypes.string.isRequired
}

export default LoginWidget;
