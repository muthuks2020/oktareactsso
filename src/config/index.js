export { default as securityConfig } from './security'
export { default as securedRoutes } from './routes'