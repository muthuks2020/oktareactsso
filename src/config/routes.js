// imports
import React from "react";
import { Lock } from "@material-ui/icons";
import { Home, Protected } from "pages";

export default [
  {
    icon: () => <Lock />,
    component: () => <Home />,
    exact: true,
    title: "Home",
    path: "/",
  },
  {
    icon: () => <Lock />,
    component: () => <Protected />,
    exact: false,
    title: "Protected",
    path: "/protected",
  },
];
