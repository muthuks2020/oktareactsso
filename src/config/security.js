const {
  REACT_APP_CLIENT_ID: clientId,
  REACT_APP_DOMAIN,
  HOST = "localhost",
  PORT = 3000,
  HTTPS = false,
} = process.env;

const issuer = REACT_APP_DOMAIN + "/oauth2/default";

const PROTOCOL = HTTPS ? "https" : "http";

const redirectUri = `${PROTOCOL}://${HOST}:${PORT}/implicit/callback`;

const scopes = ["openid", "profile", "email"];

export default {
  clientId,
  issuer,
  pkce: true,
  redirectUri,
  scopes,
};
