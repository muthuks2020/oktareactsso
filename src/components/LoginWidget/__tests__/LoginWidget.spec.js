import React from "react";
import { mount } from "enzyme";
import LoginWidget from "..";
import { Security } from "@okta/okta-react";
import { securityConfig } from "config";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

describe("<LoginWidget />", () => {
  let authService;
  let authState;
  let mockProps;
  let mockIssuer;
  beforeEach(() => {
    authState = {
      isPending: true,
    };
    authService = {
      on: jest.fn(),
      updateAuthState: jest.fn(),
      getAuthState: jest.fn().mockImplementation(() => authState),
      handleAuthentication: jest.fn(),
    };
    mockProps = { authService, ...securityConfig };
    mockIssuer = "undefined/oauth/v2";
  });

  test("should render to snapshot", () => {
    let component = renderer.create(
      <MemoryRouter>
        <Security {...mockProps}>
          <LoginWidget issuer={mockIssuer} />
        </Security>
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render two input fields and a button", () => {
    const wrapper = mount(
      <MemoryRouter>
        <Security {...mockProps}>
          <LoginWidget issuer={mockIssuer} />
        </Security>
      </MemoryRouter>
    );
    expect(wrapper.find("#password")).toBeTruthy();
    expect(wrapper.find("#username")).toBeTruthy();
    expect(wrapper.find('#submit')).toBeTruthy();
  });
});
