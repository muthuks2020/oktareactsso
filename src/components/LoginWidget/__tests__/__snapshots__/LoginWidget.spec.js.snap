// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginWidget /> should render to snapshot 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="makeStyles-paper-1"
  >
    <h1
      className="MuiTypography-root MuiTypography-h5"
    >
      Sign in
    </h1>
    <form
      className="makeStyles-form-2"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="username"
          id="username-label"
        >
          Username
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="username"
            autoFocus={true}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="username"
            name="username"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-100 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-102"
            >
              <span>
                Username
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="password"
          id="password-label"
        >
          Password
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="current-password"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="password"
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-100 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-102"
            >
              <span>
                Password
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-3 MuiButton-fullWidth"
        disabled={false}
        id="submit"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
        value="Submit"
      >
        <span
          className="MuiButton-label"
        >
          Log in
        </span>
      </button>
    </form>
  </div>
</main>
`;
